pkg/sql/tenant.go

```diff
@@ -202,6 +202,33 @@ func CreateTenantRecord(
 	)
 }
 
+func GetAllTenantNames(
+	ctx context.Context, execCfg *ExecutorConfig, txn *kv.Txn,
+) ([]roachpb.TenantName, error) {
+	if !execCfg.Settings.Version.IsActive(ctx, clusterversion.V23_1TenantNames) {
+		return nil, errors.Newf("tenant names not supported until upgrade to %s or higher is completed",
+			clusterversion.V23_1TenantNames.String())
+	}
+	rows, err := execCfg.InternalExecutor.QueryBuffered(
+		ctx, "get-tenant-names", txn,
+		`SELECT name FROM system.tenants`,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	tenants := make([]roachpb.TenantName, 0, len(rows))
+	for _, tenant := range rows {
+		// If the tenants is in the DROP state the then name is nil.
```

Comment:
I believe tenant name can also be zero if it was created prior to V23_1TenantNames, or using the `create_tenant(id)` builtin.